src/tropicalnumber.jl

```diff
@@ -0,0 +1,65 @@
+const pm_TropicalNumber_suppAddition = Union{pm_Min, pm_Max}
+const pm_TropicalNumber_suppScalar = Union{pm_Rational}
+
+#convert input to supported Scalar type
+pm_TropicalNumber{A,S}(scalar::Real) where {A <: pm_TropicalNumber_suppAddition, S <: pm_TropicalNumber_suppScalar} = pm_TropicalNumber{A,S}(convert(S,scalar))
+
+#pm_Rational es default Scalar
+pm_TropicalNumber{A}(scalar::Real) where A = pm_TropicalNumber{A,pm_Rational}(scalar)
+pm_TropicalNumber{A}() where A = pm_TropicalNumber{A,pm_Rational}()
+
+function Base.:(==)(x::pm_TropicalNumber{A,S},y::Real) where {A, S}
+    return x == pm_TropicalNumber{A,S}(y)
+end
+
+Base.:(==)(x::Real,y::pm_TropicalNumber{A,S}) where {A, S} = (y == x)
+
+#catch wrong typings for basic operations
+Base.promote_rule(::Type{<:pm_TropicalNumber{A, S1}}, ::Type{<:pm_TropicalNumber{A, S2}}) where {A, S1, S2} = pm_TropicalNumber{A, promote_type(S1,S2)}
+
+for op in (:+, :*, ://, :<)
+    @eval begin
+        Base.$(op)(x::pm_TropicalNumber{A, S}, y::pm_TropicalNumber{A, T}) where {A,S,T} = $(op)(promote(x,y)...)
```

Comment:
```suggestion
        Base.$(op)(x::pm_TropicalNumber{A}, y::pm_TropicalNumber{A}) where {A} = $(op)(promote(x,y)...)
```

Reply:
actually, this definition is not used right now. @kalmarek mentioned it would be useful to have this included in the case that we later also allow different scalar types and want to add e.g. a `pm_TropicalNumber{pm_Min,pm_Integer}` to a `pm_TropicalNumber{pm_Min,pm_Rational}`

Reply:
yeah not that important